### PR TITLE
Enhancing CoreAddon with features

### PR DESCRIPTION
**Pull request by elamaran11**
*Issue #, if available:*

*Description of changes:* Enhancing CoreAddon to support Installation of New Addons with following :
1. Making `Version` Property Optional to support upcoming Addons Such as ADOT which does have a version number parameter. Also adding a logic to make sure addon creation works with or without `version` field.
2. Adding a new field `saName` to make sure CoreAddon does not populate the AddonName with SA Name. Some Addons require a separate `saName` for the addon to work perfectly.
3. Adding a new field `namespace` field to support creation of Service Accounts in required namespace. Today all Service Accounts needs by Addons are created in `kube-system` namespace.

By submitting this pull request, I confirm that you can use, modify, copy, and redistribute this contribution, under the terms of your choice.

### CODE REVIEW

File: lib/addons/core-addon/index.ts [hunk 1]
```diff
@@ -43,27 +47,44 @@ export class CoreAddOn implements ClusterAddOn {
...
-            addonVersion: this.coreAddOnProps.version,
-            clusterName: clusterInfo.cluster.clusterName,
-            serviceAccountRoleArn: serviceAccountRoleArn,
-            resolveConflicts: "OVERWRITE"
-        });
-        
-        if (serviceAccount) {
-            cfnaddon.node.addDependency(serviceAccount);
+        if (this.coreAddOnProps.version) {
+            const cfnaddon = new CfnAddon(clusterInfo.cluster.stack, this.coreAddOnProps.addOnName + "-addOn", {
```
Comment (shapirov103) [inline lib/addons/core-addon/index.ts:65]: let's fix the code duplication on line 65-73 qne 78 - 87 as we discussed.

**Review by shapirov103 — COMMENTED** [review]
Looks great, one comment.

File: lib/addons/core-addon/index.ts (hunk 1 above)
Comment (elamaran11) [reply lib/addons/core-addon/index.ts]: This is addressed with never changes to CoreAddon to fix all issues. Please check.

**Review by shapirov103 — COMMENTED** [review]
Ready to merge. Please address the lint warning:

```
Run make lint
node node_modules/.bin/eslint . --ext .js,.jsx,.ts,.tsx

/home/runner/work/cdk-eks-blueprints/cdk-eks-blueprints/lib/addons/core-addon/index.ts
Warning:   [7](https://github.com/aws-quickstart/cdk-eks-blueprints/runs/7542731484?check_suite_focus=true#step:6:8)0:10  warning  Missing semicolon  semi

✖ 1 problem (0 errors, 1 warning)
  0 errors and 1 warning potentially fixable with the `--fix` option.
```

**Comment by elamaran11** [comment]
> Ready to merge. Please address the lint warning:
> 
> ```
> Run make lint
> node node_modules/.bin/eslint . --ext .js,.jsx,.ts,.tsx
> 
> /home/runner/work/cdk-eks-blueprints/cdk-eks-blueprints/lib/addons/core-addon/index.ts
> Warning:   [7](https://github.com/aws-quickstart/cdk-eks-blueprints/runs/7542731484?check_suite_focus=true#step:6:8)0:10  warning  Missing semicolon  semi
> 
> ✖ 1 problem (0 errors, 1 warning)
>   0 errors and 1 warning potentially fixable with the `--fix` option.
> ```

Fixed the `semi-colon` lint issue. Good to merge.

**Review by shapirov103 — APPROVED** [review]
LGTM

**Pull request merged**